### ISSUE TITLE
[Self-registration] Add number and add email action should place focus on new text field

### ISSUE THREAD

**Issue by johanna-skylight** [CDCgov/prime-simplereport#5833]
## Area or Component

PXP flow patient self-registration -> contact information section

## Quick Summary

The buttons `add another number` and `add another email` allow the user to enter extra information by dynamically adding text fields. However these new fields get added in the section above where the current focus is making this update not detected by the screen reader.

An extra item to get fixed in this page is hiding the Simple Report logo area so screen reader skips it.

### How to Replicate

Go to the self-registration form and click in either button. Verify the focus remains on the button and does not move to the new text field.

## Accessibility Violation

This issue was missed by axe dev tools but by manual testing I was able to verify the content update could get lost.

## Action Requested

Place the focus on the new text input field after it gets added.
Add `aria-hidden` to the logo image.
Recommend testing fix with screen reader to see if field is easier to find/use or that user knows they're now focused on the field

**Logo**
<img width="423" alt="Screenshot 2023-05-17 at 3 24 31 PM" src="https://github.com/CDCgov/prime-simplereport/assets/103958711/3bb214e3-7f5b-405a-836a-cb3e58d300ee">

**Dynamic Text Inputs**
<img width="515" alt="Screenshot 2023-05-17 at 3 24 45 PM" src="https://github.com/CDCgov/prime-simplereport/assets/103958711/580809fc-f1bb-49c8-90f2-91f9de75ac5b">
<img width="574" alt="Screenshot 2023-05-17 at 3 24 49 PM" src="https://github.com/CDCgov/prime-simplereport/assets/103958711/abfcdd3f-54b0-4860-bf48-ed11e4e15148">
